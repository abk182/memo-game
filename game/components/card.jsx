import React from 'react'

/*
export const Card = ({value,handler}) => (
    <div className={value+' '+'card'} onClick={(e)=>handler(value,e.target)}>
        <img src={'/img/cards/'+value+'.png'} className={'card show'}/>
    </div>
)*/


export class Card extends React.Component{
    constructor(props){
        super(props)
        this.state = {
            card_val:props.card_val,
            card_id:props.card_id,
            is_active:true
        }
    }
    componentDidMount(){
        let tmp_func = this.onCardClick;
        this.onCardClick = function () {
            alert('Rano!')
        }
        setTimeout(function () {
            this.setState({is_active:false})
            this.onCardClick = tmp_func;
        }.bind(this),3000)
    }
    componentWillReceiveProps(){
        //console.log(this.props)
        if(this.props.counter==2){this.setState({is_active:false})}
    }
    onCardClick(){
        if(!this.state.is_active && this.props.counter<2){
            this.setState({is_active:true})
            this.props.handler();
        }
    }
    render(){
        let pic = this.state.is_active ? this.state.card_val : 'hidden'
        return(
            <div className={this.props.value+' '+'card'} onClick={()=>this.onCardClick()}>
                <img src={'/img/cards/'+pic+'.png'} className={'card'}/>
            </div>
        )
    }
}