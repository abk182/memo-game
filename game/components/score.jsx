import React from 'react'
import { Link } from 'react-router-dom'

export const Score = ({score}) => (
    <div className={'score'}>
        <p>Score: {score}</p>
        <p>
            <Link to = {''}>
                end game
            </Link>
        </p>
    </div>
)