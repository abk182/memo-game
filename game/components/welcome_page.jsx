import React from 'react'
import { Link } from 'react-router-dom'

export class WelcomePage extends React.Component{
    constructor(props){
        super(props)
    }
    render(){
        return(
            <div className={'start'} data-tid="NewGame-startGame">
                <Link to = {'/table'}>
                    <button className={'start-btn'}>!112!!!!</button>
                </Link>
            </div>
        )
    }
}