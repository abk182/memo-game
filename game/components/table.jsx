import React from 'react'
import { Engine } from "../engine/engine";
import { Card } from "./card.jsx";
import { Score } from "./score.jsx";

export class GameTable extends React.Component{
    constructor(props){
        super(props)
        this.state={
            playCards:[],
            selectedCard:{},
            counter:0,
            score:0
        }
    }
    componentWillMount(){
        let engine = new Engine()
        this.setState({playCards:engine.generateCards()})
    }
    componentDidMount(){
    }
    selectCard(card){
        if(!card.state.is_open && !card.state.is_active && card.props.counter<2) {
            card.setState({is_active: true},)
            this.setState({counter:this.state.counter+1})
            if (!this.state.selectedCard.hasOwnProperty('state')){
                this.setState({selectedCard:card})
            } else {
                if (card.state.val == this.state.selectedCard.state.val){
                    this.state.selectedCard.setState({is_open:true})
                    card.setState({is_open:true})
                    this.setState({selectedCard:{},counter:0,score:this.state.score+1})
                } else {
                    setTimeout(function () {
                        this.state.selectedCard.setState({is_active:false})
                        card.setState({is_active:false})
                        this.setState({selectedCard:{},counter:0,score:this.state.score-1})
                    }.bind(this),500)
                }
            }
        }

    }
    render(){
        return(
            <div className='table'>
                {
                    this.state.playCards.map(card=>(
                        <Card key={card.id}
                              card_id={card.id}
                              card_val={card.value}
                              counter={this.state.counter}
                              handler={(obj)=>this.selectCard(obj)}/>
                    ))
                }
                <Score score={this.state.score}/>
            </div>
        )
    }
}



