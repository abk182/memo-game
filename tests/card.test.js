import React from 'react'
import renderer from 'react-test-renderer'
import { Card } from '../game/components/card'

test('Card test is working', () => {
    const component = renderer.create(
        <Card id={'01'} value={'H5'} handler={obj=>console.log(obj)}/>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.props.onClick()

    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})