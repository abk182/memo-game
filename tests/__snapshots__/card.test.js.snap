// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test is working 1`] = `
<div
  className="card"
  data-tid="Card"
  onClick={[Function]}
>
  <img
    src="/img/cards/undefined.png"
  />
</div>
`;

exports[`Card test is working 2`] = `
<div
  className="card"
  data-tid="Card"
  onClick={[Function]}
>
  <img
    src="/img/cards/undefined.png"
  />
</div>
`;
